api/app.py

```diff
@@ -14,19 +14,68 @@
 
 mysql = MySQL(app)
 
+@app.route('/api/building/create', methods = ["POST"])
+def create_building():
+    # info is a dictionary with keys address, pet_friendly, laundry_availability, type_of_unit, and distance_from_waterloo
+    # address VARCHAR(255) NOT NULL,
+    # pet_friendly TINYINT(1) NOT NULL,
+    # laundry_availability ENUM('building', 'ensuite', 'none') NOT NULL,
+    # type_of_unit ENUM('apartment','house') NOT NULL,
+    # distance_from_waterloo DECIMAL(3,1) NOT NULL,
 
+    building_info = request.get_json() # {"address": "123 H", "pet_friendly": 0, "laundry_availability": "ensuite", "type_of_unit": "house", "distance_from_waterloo": 0.4}
+    conn = mysql.connection
+    cur = conn.cursor()
 
-@app.route('/api')
-def get_message():
-    return {'message': 'Hello from the API'}
+    try:
+        cur.execute("INSERT INTO Building VALUES (NULL, %s, %s, %s, %s, %s)", 
+                    (building_info["address"], building_info["pet_friendly"], building_info["laundry_availability"], building_info["type_of_unit"], building_info["distance_from_waterloo"]))
+        cur.close()
+        conn.commit()
+        return {"status": True}
+    except Exception as e:
+        return {"status": False, "message": f"Error with inserting: {e}"} 
```

Comment:
can you also add an error status code (the default is set to 200)

Reply:
wdym, what should all the possible ret values be

Reply:
It doesn't really make much of a difference, but typically a 200 status code is given when the request is ok. In this case, the request doesn't go through, we set the status code to false, but the status code is still 200. It's more of a design thing, but it's better to be consistent.

Reply:
we can add a generic response code, is 400 okay?

Reply:
ya